Lib/importlib/_bootstrap_external.py

```diff
@@ -1092,8 +1092,17 @@ class ExtensionFileLoader(FileLoader, _LoaderBasics):
 
     """
 
-    def __init__(self, name, path):
-        self.name = name
+    def __init__(self, fullname=None, path=None, *, name=None):
+        if name is not None and fullname is not None:
+            raise TypeError("'name' and 'fullname' cannot be used "
+                            "simultaneously ('fullname' is preferred)")
```

Comment:
We can drop the "('fullname' is preferred)" part if we emit `DeprecationWarning` before raising the exception.

```py
if name is not None:
    _warnings.warn(...)
    if fullname is not None:
        raise TypeError(...)
    ...
```

Reply:
Of course, we could keep this logic and tweak the exception message instead.

I'm fine with both approaches as long as the user knows `name` is deprecated.